### PR TITLE
 igir: use autoPatchelfHook

### LINKED ISSUE #279695: igir ships with prebuild 7zip that won't directly run on nixos
### Describe the bug
When letting igir access some 7z files it tries to execute a bundled binary

### Steps To Reproduce
1. copy a 7z file somewhere
2. run igir with --dat and --input to the path where the 7z is.
3. igir should try to extract the 7z to create the checksum of it's contents but this failes.

### Expected behavior
It should depend and use 7zip deployed through nix

### Screenshots
```
 ______   ______   ______  _______
|      \ /      \ |      \|       \
 \$$$$$$|  $$$$$$\ \$$$$$$| $$$$$$$\
  | $$  | $$|    \  | $$  | $$    $$   ROM collection manager
  | $$  | $$|    \  | $$  | $$    $$   https://igir.io/
  | $$  | $$ \$$$$  | $$  | $$$$$$$\
 _| $$_ | $$__| $$ _| $$_ | $$  | $$   v2.2.0
|   $$ \ \$$    $$|   $$ \| $$  | $$
 \$$$$$$  \$$$$$$  \$$$$$$ \$$   \$$


NOTICE: An update is available for igir: v2.2.1
✓ Scanning for DATs ·········· | 27 unique DATs found
ERROR: xyz.7z: failed to parse file: 7-zip exited with code 127
ERROR:    Could not start dynamically linked executable:
/nix/store/19yb88k8q8ymrdiky0j6dnng1knnvpl3-igir-2.2.0/lib/node_modules/igir/node_modules/7zip-bin/linux/x64/7za
ERROR:    NixOS cannot run dynamically linked executables intended for generic
ERROR:    linux environments out of the box. For more information, see:
ERROR:    https://nix.dev/permalink/stub-ld
```

### Notify maintainers
@TheBrainScrambler

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TheBrainScrambler**
This fixes https://github.com/NixOS/nixpkgs/issues/279695

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TheBrainScrambler** [comment]
Result of `nixpkgs-review pr 279842` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>igir</li>
  </ul>
</details>

**Comment by NickCao** [comment]
I don't think this is a correct fix. The build output does not reference p7zip.

```
❯ nix path-info -Srh ./result
/nix/store/i93s7xxblavsacpy82zdbn4kplsyq48l-libunistring-1.1       1.8M
/nix/store/ji01n9vinnj22nbrb86nx8a1ssgpilx8-libidn2-2.3.4          2.1M
/nix/store/ldrslljw4rg026nw06gyrdwl78k77vyq-xgcc-12.3.0-libgcc   139.1K
/nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27         31.1M
/nix/store/l3lhzwrfgzncjj1z4wyq6kkrp6czx5qp-gcc-12.3.0-libgcc    139.1K
/nix/store/dghjv6hfz0s0z4kffa5ahyw2mhp79215-gcc-12.3.0-lib        38.8M
/nix/store/42xvz6zy0n46ihc4qbz3swmphrnnl7j5-gmp-with-cxx-6.3.0    39.5M
/nix/store/c7rf6w8xvxc4423m9rsgq5nsqhknshvy-openssl-3.0.12        37.3M
/nix/store/q8qq40xg2grfh9ry1d9x4g7lq4ra7n81-bash-5.2-p21          32.6M
/nix/store/sr9ka27mzwb3mpvmh1k9180k9rrzlrky-openssl-3.0.12-bin    39.8M
/nix/store/7yx2i1wm89zhypky0l0lmmpl10yf0bh7-openssl-3.0.12-dev    41.7M
/nix/store/v3kjfcaysin90dqyna1rwgnxsf3pg7fz-attr-2.5.1            31.2M
/nix/store/7zqx95qyg38fic8mzk0nxz06ag3jkh12-acl-2.3.1             31.3M
/nix/store/l77dphhwcw7j5k5kqxw1rqxbvhszgaw7-icu4c-73.2            78.0M
/nix/store/zx8aqgdy735qzk81glfwil6mbi6ddqb1-coreutils-9.4         41.1M
/nix/store/9dhsn7y43f0s1vvxwby671n9qb8ab4vm-icu4c-73.2-dev        85.8M
/nix/store/lprrvg44cyn9ajpq77lh7rlnp4kzdqzs-zlib-1.3              31.2M
/nix/store/fak09pxs42fqcgngf2yznw83f88kmw7s-zlib-1.3-dev          31.3M
/nix/store/p4x3hr44ah240zascb7ygk4m5b6sbrnx-libuv-1.47.0          31.3M
/nix/store/v0r02wg7qbxchccl81k1x1srxg9nb9ib-libuv-1.47.0-dev      31.4M
/nix/store/yc5bicc5zs5czjp5q5aygz79km84rmd4-nodejs-20.10.0       160.7M
/nix/store/qcav845wwqj12sjla9zfx7hjn83laf3j-igir-2.2.1           201.9M
```

**Comment by TheBrainScrambler** [comment]
Ah thanks, indeed ... I tested `igir` on a random 7z of mine before doing this PR and I was sure it worked, as I remember having a different error that doesn't say "Could not start dynamically linked executable" ... Except that this is the error I'm getting now. The only explanation i see is that I went too fast on it and didn't see that error. So this problem might be more complicated to solve than I thought ...

Do you happen to have a proper DAT collection that's in a 7z file, so I can test this properly ?

**Comment by TheBrainScrambler** [comment]
Ok, after thinking long enough about it the solution is quite simple: I just use `autoPatchelfHook` to patch the bundled binary so that you can use it.

I've now tested it properly, after all I could just make myself a 7z that contains a DAT from the DATs I have ... It works.

File: pkgs/by-name/ig/igir/package.nix [hunk 1]
```diff
@@ -21,6 +26,14 @@ buildNpmPackage rec {
     patchShebangs scripts/update-readme-help.sh
   '';
 
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [ stdenv.cc.cc.lib ];
+
+  # from lib/node_modules/igir/node_modules/@node-rs/crc32-linux-x64-musl/crc32.linux-x64-musl.node
+  # I don't know what "crc32.linux-x64-musl.node" is
```
Comment (NickCao) [inline pkgs/by-name/ig/igir/package.nix:34]: You can drop this line. It's just some native extension of that node module, but for the musl libc, irrelevant for our use here.

Comment (TheBrainScrambler) [reply pkgs/by-name/ig/igir/package.nix]: Ah ok, I've changed it to say that it's irrelevant for us

**Review by NickCao — APPROVED** [review]

**Pull request merged**